src/InputGroup/_InputGroupAddon.js

```diff
@@ -0,0 +1,45 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const InputGroupAddon = ({
+    addonClassNames,
+    children,
+    className,
+    compact,
+    isButton,
+    ...otherProps
+}) => {
+
+    const addonClasses = classnames(
+        addonClassNames,
+        'fd-input-group__addon',
+        { 'fd-input-group__addon--button': isButton },
+        { 'fd-input-group__addon--button--compact': compact && isButton }
+    );
+
+    return (
+        <span className={addonClasses} {...otherProps} >
+            {children}
+        </span>
+    );
+};
+
+InputGroupAddon.displayName = 'InputGroup.Addon';
+
+InputGroupAddon.propTypes = {
+    addonClassNames: PropTypes.string,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    compact: PropTypes.bool,
```

Comment:
There are no defined CSS classes for a "compact" addon.  This should be passed to its children (if we keep the prop).  See comment in `InputGroup.js` about how to handle `compact` prop.